v2/errors.go

```diff
@@ -0,0 +1,28 @@
+package car
+
+import (
+	"fmt"
+)
+
+type (
+	// ErrCidTooLarge signals that a CID is too large to include in CARv2 index.
+	ErrCidTooLarge error
```

Comment:
not sure I get the point of this error type. Either export errCidTooLarge with its fields, or since NewErrCidTooLarge is exported, or unexport NewErrCidTooLarge and just use `error` types as return parameters.